roles/openshift_inventory/tasks/main.yml

```diff
@@ -0,0 +1,22 @@
+---
+- no_log: True  # keep task descriptions legible
+  block:
+
+  - name: Standardize on latest variable names
+    set_fact:
+      deployment_type: "{{ openshift_deployment_type | default(deployment_type) | default(None) }}"
+      openshift_deployment_type: "{{ openshift_deployment_type | default(deployment_type) | default(None) }}"
+
+  - name: Normalize openshift_release
+    set_fact:
+      openshift_release: "{{ openshift_release | string | regex_replace('^v', '') }}"
+    when: openshift_release is defined
+
+- name: Ensure a valid deployment type has been given |
+  when: >
+    openshift_deployment_type is not defined
+    or openshift_deployment_type not in known_openshift_deployment_types
+  fail:
+    msg: |-
+      Please set openshift_deployment_type to one of:
+      {{ known_openshift_deployment_types | join(', ') }}
```

Comment:
I'm curious about your use of the YAML `>` character for indicating *folded-style* string content in this `when`. That's just so you could write the condition of the `when` across multiple lines, right?

And the `|-` character here I am not 100% certain of the exact side-effects. I understand that the `|` pipe character is used to indicate a literal string in which new-line characters **are** preserved. I am not sure what the `-` hyphen after the pipe indicates.